app/controllers/users_controller.rb

```diff
@@ -24,8 +28,57 @@ def create
     end
   end
 
+  def edit; end
+
+  def update
+    if @user.update user_params
+      flash[:success] = I18n.t("layouts.messages.updated_success")
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user.destroy
+      flash[:success] = I18n.t("layouts.messages.user_deleted")
+      redirect_to root_path
+    else
+      flash[:danger] = I18n.t("layouts.messages.deleted_fail")
+      redirect_to user_path
+    end
+  end
+
   private
 
+  def admin_user
+    redirect_to root_path unless current_user.admin?
+    flash[:warning] = t("layouts.messages.admin_error")
```

Comment:
ri thì flash ở lien 56 lúc nào cũng có à cu
```suggestion
    return if current_user.admin?
    redirect_to root_path  
    flash[:warning] = t("layouts.messages.admin_error")
```